cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -598,24 +602,156 @@ private predicate argumentFlowsThrough(
 ) {
   nodeCand1(out, unbind(config)) and
   not inBarrier(out, config) and
-  compatibleTypes(t2, getErasedNodeType(out)) and
+  compatibleTypes(t2, getErasedNodeTypeBound(out)) and
   exists(DataFlowCall call, ReturnKindExt kind |
     argumentFlowsThrough0(call, arg, kind, t1, t2, summary, config) and
     out = kind.getAnOutNode(call)
   )
 }
 
+private newtype TNodeExt =
+  TNormalNode(Node node) { nodeCand1(node, _) } or
+  TReadStoreNode(DataFlowCall call, Node node1, Node node2, Content f1, Content f2) {
+    exists(Configuration config |
+      nodeCand1(node1, config) and
+      argumentValueFlowsThrough(call, node1, TContentSome(f1), TContentSome(f2), node2) and
+      nodeCand1(node2, unbind(config))
```

Comment:
We should add the `readStoreCand1(f1, unbind(config)) and readStoreCand1(f2, unbind(config))` here.